### DISCUSSION TITLE
How can you search for Dates in a TextColumn?

### DISCUSSION THREAD

**Discussion by MACscr** [filamentphp/filament#3398] [Help]
If I am using something like:

```php
TextColumn::make('opening_date')
    ->dateTime('n/j/y g:ia')
    ->searchable()
    ->sortable(),
```

How can i even search for the date? Seems i can't search for `8/2/22` or whatever and find anything that matches. Do i really have to setup a virtual field for this? Would be nice if there was a simpler solution.

**Comment by danharrin** [comment]
How would you do it with Eloquent outside of Filament?

**Reply by MACscr** [reply]
I have no idea. Its not something i have tackled before. I figured you all had already. =P

**Reply by danharrin** [reply]
You'd likely have to use Scout & index a few different human-readable date formats in `toSearchableArray()`. I doubt MySQL is powerful enough

**Comment by MACscr** [comment]
Why a few? Im only wanting the user to be able to search for a date by the same format it’s being displayed to the user. Seems rather reasonable. This could maybe be solved by simply being able to set that format in searchable()?

- Mark

**Reply by danharrin** [reply]
No, because afaik MySql cannot search using a PHP date format. We do not use PHP to search at all
